### DISCUSSION TITLE
Safely catch data from ajax

### DISCUSSION THREAD

**Discussion by VVStz** [pallets/flask#5210] [Q&A]
Hello,

I'm sending data (python class, `list`, `dict`) trough ajax to Flask.

What would be the best way to retrieve them ?

I've seen in the [documentation](https://flask.palletsprojects.com/en/2.3.x/api/#flask.request)

That prior to use `request.get_data` we should check the data length.
>Usually it’s a bad idea to call this method without checking the content length first as a client could send dozens of megabytes or more to cause memory problems on the server.

is it correct that `request.content_length` retrieve the information from the header fields ?

Is [MAX_CONTENT_LENGTH](https://flask.palletsprojects.com/en/2.3.x/config/#MAX_CONTENT_LENGTH) trigger before `request.get_data` ? and therefore is it enough ?

Thank you.

**Comment by tachyondecay** [comment]
If you send your request as JSON, then your data will be available in` request.json` . If you send your request as an ordinary POST request, the data will be in `request.form`

See this page for more info: https://flask.palletsprojects.com/en/2.3.x/patterns/javascript/
